core/engine_test.go

```diff
@@ -690,8 +690,14 @@ func TestRunTags(t *testing.T) {
 	t.Parallel()
 	tb := httpmultibin.NewHTTPMultiBin(t)
 
-	runTagsMap := map[string]string{"foo": "bar", "test": "mest", "over": "written"}
-	runTags := metrics.NewSampleTags(runTagsMap)
+	runTags := map[string]string{"foo": "bar", "test": "mest", "over": "written"}
+
+	// it copies the map so in the case the runner will overwrite
+	// some run tags' values it doesn't affect the assertion.
+	expectedRunTags := make(map[string]string)
+	for k, v := range expectedRunTags {
```

Comment:
```suggestion
	for k, v := range runTags {
```

I am pretty sure this should be `runTags`, but not why the test didn't fail? :sweat_smile:

Reply:
Because the line below goes over `expectedRunTags` and checks that it's key,values are in the final tags - which when it's empty is always true

Reply:
I have reversed them so it should be safer now